pkg/test/suite/run.go

```diff
@@ -156,6 +157,13 @@ func runTestCaseWithSharedEnvironment(t *testing.T, suite TestingSuite, suiteOpt
 		stream.ResetInMemoryOutputs()
 		stream.ResetProducerDaemons()
 		kvstore.ResetConfigurableKvStores()
+
+		disp, err := dispatcher.ProvideDispatcher(suite.Env().Context(), suite.Env().Config(), suite.Env().Logger())
```

Comment:
maybe extract `suite.Env()` into its own variable if you call it multiple times?